### PR TITLE
💄 Update airdrop P2 UI 

### PR DESCRIPTION
**Pull request by AuroraHuang22**

### CODE REVIEW

File: pages/airdrop.vue [hunk 1]
```diff
@@ -201,7 +201,7 @@ const signerModule = namespace('signer')
         {
           hid: 'og:image',
           property: 'og:image',
-          content: 'https://app.like.co/images/og/airdrop_launch.png',
+          content: 'https://app.rinkeby.like.co/images/og/airdrop_launch.png',
```
Comment (nwingt) [inline pages/airdrop.vue:204]: The endpoint should be determined by environment

File: pages/airdrop/claim.vue [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+  >
+    <img
+      :class="[
+        'mt-[32px]',
+        'w-[300px]',
+        'mx-auto',
+      ]"
+      src="/images/airdrop/title_Claim.png"
+    />
+    <AirdropDashborad
```
Comment (williamchong) [inline pages/airdrop/claim.vue:21]: AirdropDashboard

File: pages/airdrop/claim.vue [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+        const res: any = await this.$axios
+          .get(`${AIRDROP_CLAIM}${this.currentAddress}`)
+          .catch((err) => {
+            if (err.response)
+              this.errorMessage = this.$t(
+                'AirDrop.errorMessage.noAddress',
+              ) as string
+          })
+        this.totalAirdrop = Math.round(
+          (res.data.reward.unclaimedAmount + res.data.reward.claimedAmount) *
```
Comment (williamchong) [inline pages/airdrop/claim.vue:168]: Should use bignumber to handle

File: pages/airdrop/claim.vue [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+    if (this.currentAddress) {
+      try {
+        const res: any = await this.$axios
+          .get(`${AIRDROP_CLAIM}${this.currentAddress}`)
+          .catch((err) => {
+            if (err.response)
+              this.errorMessage = this.$t(
+                'AirDrop.errorMessage.noAddress',
+              ) as string
+          })
```
Comment (williamchong) [inline pages/airdrop/claim.vue:166]: Why not handle in catch block below?

File: pages/airdrop/claim.vue [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+    try {
+      const res: any = await this.$axios
+        .post(
+          `${AIRDROP_MISSION}${this.currentMission.name}?address=${this.currentAddress}`,
+        )
+        .catch((err) => {
+          console.log(err.response)
+        })
+      this.currentMission.claimedAmount = Math.round(
+        res.data.claimedAmount * Denom.Nanolike,
```
Comment (williamchong) [inline pages/airdrop/claim.vue:268]: Should use bignumber to handle

File: pages/airdrop/claim.vue [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+  async handleMissionDone() {
+    this.loadingStatus = 'Loading'
+    try {
+      const res: any = await this.$axios
+        .post(
+          `${AIRDROP_MISSION}${this.currentMission.name}?address=${this.currentAddress}`,
+        )
+        .catch((err) => {
+          console.log(err.response)
+        })
```
Comment (williamchong) [inline pages/airdrop/claim.vue:266]: Should use `console.error`, why not use the catch block below

File: pages/airdrop/claim.vue [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+      this.currentMission.claimedAmount = Math.round(
+        res.data.claimedAmount * Denom.Nanolike,
+      )
+      this.currentMission.isCompleted = true
+      this.currentMission.txHash = res.data.txHash
+    } catch (error) {
+      console.log(error)
```
Comment (williamchong) [inline pages/airdrop/claim.vue:273]:
```suggestion
      console.error(error)
```

File: components/AirdropMissionDialog.vue [hunk 8]
```diff
@@ -0,0 +1,364 @@
...
+  @Prop(Object) readonly mission!: object | {}
+
+  buttonText: string = this.$t('AirDrop.mission.button.completed') as string
+
+  changeText(status: string) {
+    if (status === 'in')
+      this.buttonText = this.$t(
+        'AirDrop.mission.button.seeTransaction',
+      ) as string
+    else this.buttonText = this.$t('AirDrop.mission.button.completed') as string
```
Comment (williamchong) [inline components/AirdropMissionDialog.vue:361]: Add `{}` for multiline if else

**Review by williamchong — CHANGES_REQUESTED** [review]

File: pages/airdrop/claim.vue [hunk 9]
```diff
@@ -158,19 +155,14 @@ export default class AirdropClaimPage extends Vue {
...
-        this.totalClaimedAmount = Math.round(
-          res.data.reward.claimedAmount * Denom.Nanolike,
-        )
+
+        this.totalAirdrop = new BigNumber(
+          res.data.reward.unclaimedAmount + res.data.reward.claimedAmount,
+        ).shiftedBy(-9).integerValue() 
+        this.totalClaimedAmount = new BigNumber(
+          res.data.reward.claimedAmount,
+        ).shiftedBy(-9).integerValue()
```
Comment (williamchong) [inline pages/airdrop/claim.vue]: Are these expected to be 100% integer? Should we use round_down mode?

File: pages/airdrop/claim.vue [hunk 10]
```diff
@@ -0,0 +1,320 @@
...
+  handleMissionClose() {
+    if (!this.errorMessage) {
+      this.fetchMissionStauts()
+    }
+    this.isOpenMissionDialog = false
+    this.step = 1
+  }
+  
+  initClaimStatus() {
+    this.unclaimedAmount = new BigNumber(0)
```
Comment (williamchong) [inline pages/airdrop/claim.vue:275]: Where is this `unclaimedAmount` used?

**Review by williamchong — CHANGES_REQUESTED** [review]

File: pages/airdrop/claim.vue (hunk 9 above)
Comment (AuroraHuang22) [reply pages/airdrop/claim.vue]: Think we keep 100% integer is better for UI.

Comment (williamchong) [reply pages/airdrop/claim.vue]: Should use `BigNumber.ROUND_DOWN` mode I guess

File: components/AirdropDashboard.vue [hunk 11]
```diff
@@ -0,0 +1,197 @@
...
+export default class AirdropDashboard extends Vue {
+
+  // The total claimable amount of airdrop
+  @Prop(Object) readonly totalAirdrop!: object | undefined
+
+  // The total claimed amount of airdrop
+  @Prop(Object) readonly totalClaimedAmount!: object | undefined
+
+  // Contains the time information about decay
+  @Prop(Object) readonly decay!: object | undefined
```
Comment (williamchong) [inline components/AirdropDashboard.vue:195]: I think you are relying on the implicit `toString()` method of bignumber, a safer way to use it is call `toFixed()` or `integerValue()`. You can either call them outside this component and pass them as `number`/`string`, or keep passing them as bignumber object, and call these function in template/computed

Comment (AuroraHuang22) [reply components/AirdropDashboard.vue]: I see, I thought I had to keep the amount as bignumber object all the time.

**Review by williamchong — APPROVED** [review]

File: components/AirdropMissionDialog.vue [hunk 12]
```diff
@@ -0,0 +1,380 @@
...
+          </div>
+        </div>
+      </div>
+      
+      <!-- result -->
+      <div
+        v-if="
+          step === 3 &&
+          (claimStatus === 'unable' ||
+            (claimStatus !== 'unable' && !mission.isCompleted))"
```
Comment (nwingt) [inline components/AirdropMissionDialog.vue:213]: can refactor this into computed property

File: components/icon/MissionNotCompleted.vue [hunk 13]
```diff
@@ -0,0 +1,20 @@
+<template>
```
Comment (nwingt) [inline components/icon/MissionNotCompleted.vue:1]: The filename should be `MissionIncompleted.vue`

File: pages/airdrop.vue [hunk 14]
```diff
@@ -190,10 +211,13 @@ const signerModule = namespace('signer')
 export default class AirdropCheckPage extends Vue {
   @signerModule.Getter('getAddress') currentAddress!: string
 
+  get shouldShowSubscriptionCar() {
```
Comment (nwingt) [inline pages/airdrop.vue:214]:
```suggestion
  get shouldShowSubscriptionCard() {
```

File: pages/airdrop/index.vue [hunk 15]
```diff
@@ -1,10 +1,86 @@
...
+    ]"
+  >
+    <img
+      :class="[
+        'hidden',
+        'my-[32px]',
+        'w-[300px]',
+        'lg:block',
+      ]"
+      src="/images/airdrop/title_Claim.png"
```
Comment (nwingt) [inline pages/airdrop/index.vue:20]: Use assets relative path

**Review by nwingt — CHANGES_REQUESTED** [review]
For those mission images, I suggest to use SVG to compose multi-layers graphics,
Layer 1: BG of complete or incomplete (grayscale)
Layer 2: Images of #1, #2, #3, #4
Layer 3: `COMPLETED` text as image

**Comment by nwingt** [comment]
Should adjust the border-radius
image

File: pages/airdrop.vue (hunk 14 above)
Comment (AuroraHuang22) [reply pages/airdrop.vue:214]: 😅  🚗

**Comment by nwingt** [comment]
Should make these 2 buttons in the same width
https://user-images.githubusercontent.com/17264716/154004695-46b0acb8-ec17-4baa-9f31-594118aeaede.mov

**Pull request merged**